### DISCUSSION TITLE
Using grblHAL on a bare processor

### DISCUSSION THREAD

**Discussion by phil-barrett** [terjeio/grblHAL#135] [General]
These instructions are for the Teensy 4.1 port but are generally applicable to anyone wanting to test before installing the breakout board in a CNC machine with input switches and limit sensors. Because the various inputs (Limits and control) are typically pulled high and grblHAL defaults to normally closed inputs, the initial start up state will be ALARM. This is a common question from my customers so I wrote this blog to help them.  Suggestions and critiques are welcome.

[Blog Post](https://www.grbl.org/single-post/naked-teensy-party)

**Comment by dresco** [comment]
Hi Phil, 

Nice write-up. Is it worth giving CNCjs a mention as another sender option for non-Windows users? I've not seen any issues running against compatibility level 0.

**Comment by phil-barrett** [comment]
Thanks.  I will add it.
